DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTaskSwizzler.swift

```diff
@@ -44,7 +44,14 @@ internal final class URLSessionTaskSwizzler {
         private let method: Method
 
         static func build() throws -> TaskResume {
-            return try TaskResume(selector: self.selector, klass: URLSessionTask.self)
+            // RUM-2690: We swizzle private `__NSCFLocalSessionTask` class as it appears to be uniformly used
+            // in iOS versions 12.x - 17.x. Swizzling the public `URLSessionTask.resume()` doesn't work in 12.x and 13.x.
+            // See https://github.com/DataDog/dd-sdk-ios/pull/1637 for full `URLSessionTask` class dumps in major iOS versions.
+            let className = "__NSCFLocalSessionTask"
+            guard let klass = NSClassFromString(className) else {
+                throw InternalError(description: "Failed to swizzle `URLSessionTask`: `\(className)` class not found.")
+            }
```

Comment:
Indeed, we should start investigating as soon as we get this telemetry error. It will most likely be a change on a new OS version so we should be able to dump the object while debugging. 
Maybe we should start having monitors of specific error!